### PR TITLE
Make hpp operator CRD 'explainable'

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

Update CRD definition to be explainable. Aka add descriptions to all the fields.

### CODE REVIEW

File: deploy/operator.yaml [hunk 1]
```diff
@@ -206,97 +206,106 @@ roleRef:
...
+            type: object
+          spec:
+            description: HostPathProvisionerSpec defines the desired state of HostPathProvisioner
+            properties:
+              imagePullPolicy:
+                description: ImagePullPolicy is the container pull policy for the host
+                  path provisioner containers
+                type: string
+              pathConfig:
+                description: PathConfig is the path configuration of the path used to
```
Comment (aglitke) [inline deploy/operator.yaml:246]: PathConfig describes the location and layout of PV storage on nodes. ?

File: deploy/operator.yaml [hunk 2]
```diff
@@ -206,97 +206,106 @@ roleRef:
...
                 properties:
-                  lastHeartbeatTime:
-                    format: date-time
-                    type: string
-                  lastTransitionTime:
-                    format: date-time
-                    type: string
-                  message:
+                  path:
+                    description: Path The path the directories for the PVs are created
```
Comment (aglitke) [inline deploy/operator.yaml:250]: Maybe change to...
The provisioner will store PVs at this location on each node.

File: deploy/operator.yaml [hunk 3]
```diff
@@ -206,97 +206,106 @@ roleRef:
...
+                      under
                     type: string
-                  reason:
+                  useNamingPrefix:
+                    description: UseNamingPrefix Use the name of the PVC requesting
```
Comment (aglitke) [inline deploy/operator.yaml:254]: Does it have to start with the name of the field?  If so, it should be lowercase?  If not, consider changing to...
Indicates whether the name of the requesting PVC is included in the directory name when dynamically provisioning a PV.

File: deploy/operator.yaml [hunk 4]
```diff
@@ -206,97 +206,106 @@ roleRef:
...
+                      description: ConditionType is the state of the operator's reconciliation
+                        functionality.
+                      type: string
+                  required:
+                  - status
+                  - type
+                  type: object
+                type: array
+              observedVersion:
+                description: ObservedVersion The observed version fo the hostpath provisioner
```
Comment (aglitke) [inline deploy/operator.yaml:293]: s/fo/of/

File: deploy/operator.yaml (hunk 1 above)
Comment (awels) [reply deploy/operator.yaml:246]: Done

File: deploy/operator.yaml (hunk 2 above)
Comment (awels) [reply deploy/operator.yaml:250]: Done

File: deploy/operator.yaml (hunk 3 above)
Comment (awels) [reply deploy/operator.yaml:254]: Done

File: deploy/operator.yaml (hunk 4 above)
Comment (awels) [reply deploy/operator.yaml]: Done

File: deploy/operator.yaml [hunk 5]
```diff
@@ -206,97 +206,105 @@ roleRef:
...
+              imageTag:
+                type: string
+                enum:
+                - Always
+                - IfNotPresent
+                - Never
+              imageRegistry:
+                type: string
+              pathConfig:
+                description: describes the location and layout of PV storage on nodes create PVs
```
Comment (aglitke) [inline deploy/operator.yaml:245]: ?  spurious 'create PVs'

**Pull request merged**